src/printer-markdown.js

```diff
@@ -292,13 +294,21 @@ function genericPrint(path, options, print) {
           ]);
       }
     case "definition":
-      return concat([
-        "[",
-        node.identifier,
-        "]: ",
-        printUrl(node.url),
-        node.title === null ? "" : ` ${printTitle(node.title)}`
-      ]);
+      return group(
+        concat([
+          concat(["[", node.identifier, "]:"]),
+          indent(
+            concat(
+              [
+                line,
+                printUrl(node.url),
+                node.title ? line : null,
+                node.title ? printTitle(node.title) : null
+              ].filter(x => x)
```

Comment:
I think it'd be better this way as it's more intuitive:

```js
concat([
  line,
  printUrl(node.url),
  node.title === null
    ? ""
    : concat([line, printTitle(node.title)])
])
```

And also we should add a test case for empty title:

```md
[something]: http://prettier.io 
```

Reply:
It seems like the `""` can be 🔥d when the title is blank. Markdown (at least CommonMark) [doesn’t render links with empty titles differently from links without titles](http://spec.commonmark.org/dingus/?text=%5Bfoo%5D%0A%5Bbar%5D%0A%5Bbaz%5D%0A%5Bquux%5D%0A%0A%5Bfoo%5D%3A%20https%3A%2F%2Fexample.com%20%27%27%0A%5Bbar%5D%3A%20https%3A%2F%2Fexample.com%20%22%22%0A%5Bbaz%5D%3A%20https%3A%2F%2Fexample.com%20()%0A%5Bquux%5D%3A%20https%3A%2F%2Fexample.com%0A).

Reply:
It'll change the AST (at least `remark`'s), we shouldn't touch it.

Reply:
...sorry, I thought `node.title` was `""` in that case, it seems they're all parsed as `null`. But results are the same anyway.